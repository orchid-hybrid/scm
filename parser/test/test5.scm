(define (collector)
  ;; collect elements into a list and ask for its value
  ;; 
  ;; (let ((c (collector)))
  ;;    (c 'a) (c 'b) (c))
  (let ((list '())
        (last-cell #f))
    (lambda args
      (if (null? args)
          list
          (if last-cell
              (begin
                (set-cdr! last-cell (cons (car args) '()))
                (set! last-cell (cdr last-cell)))
              (begin
                (set! list (cons (car args) '()))
                (set! last-cell list)))))))
