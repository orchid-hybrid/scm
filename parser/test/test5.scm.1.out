(define (collector)
  (let ((list '()) (last-cell #f))
    (lambda args
      (if (null? args)
        list
        (if last-cell
          (begin
            (set-cdr! last-cell (cons (car args) '()))
            (set! last-cell (cdr last-cell)))
          (begin (set! list (cons (car args) '())) (set! last-cell list)))))))
